### PR TITLE
RUMM-1100 Generate and use the UserInfo model from json schema

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: dd-sdk-android/src/main/kotlin/com/datadog/android/log/internal/user/DatadogUserInfoProvider.kt [hunk 1]
```diff
@@ -7,16 +7,18 @@
...
 ) : MutableUserInfoProvider {
 
-    private var internalUserInfo = UserInfo()
+    private var internalUserInfo =
+        UserInfo()
         set(value) {
-            field = value
-            consentAwareWriter.write(field)
-        }
+                field = value
```
Comment (0xnm) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/log/internal/user/DatadogUserInfoProvider.kt:19]: looks like wrong indent

**Review by 0xnm — CHANGES_REQUESTED** [review]
just a typo and some formatting

File: dd-sdk-android/src/test/kotlin/com/datadog/android/tracing/internal/domain/SpanSerializerTest.kt [hunk 2]
```diff
@@ -118,7 +118,8 @@ internal class SpanSerializerTest {
         @LongForgery serverOffsetNanos: Long
     ) {
         // Given
-        val fakeMinimalUserInfo = UserInfo()
+        val fakeMinimalUserInfo =
```
[the remark this answers is not in the document]
Comment (mariusc83) [reply dd-sdk-android/src/test/kotlin/com/datadog/android/tracing/internal/domain/SpanSerializerTest.kt:121]: yup, normally it should come from `ktlint` auto - formatting but I will double check.

File: dd-sdk-android/src/test/kotlin/com/datadog/android/log/internal/domain/LogSerializerTest.kt [hunk 3]
```diff
@@ -318,19 +320,22 @@ internal class LogSerializerTest {
             if (info.id.isNullOrEmpty()) {
                 doesNotHaveField(LogAttributes.USR_ID)
             } else {
-                hasField(LogAttributes.USR_ID, info.id)
+                hasField(LogAttributes.USR_ID, info.id!!)
```
Comment (xgouchet) [inline dd-sdk-android/src/test/kotlin/com/datadog/android/log/internal/domain/LogSerializerTest.kt:323]: Why did you add the `!!`? It doesn't add anything, and the test will fail with a NPE instead of AssertException.

**Review by xgouchet — CHANGES_REQUESTED** [review]

File: dd-sdk-android/src/test/kotlin/com/datadog/android/log/internal/domain/LogSerializerTest.kt (hunk 3 above)
Comment (mariusc83) [reply dd-sdk-android/src/test/kotlin/com/datadog/android/log/internal/domain/LogSerializerTest.kt:323]: It's actually ktlint or kotlint compiler complaining about that. Basically the new NetworkInfo class is public and this test class is Internal and it thinks that this variable can be changed from outside the package. I will see if I can get rid of this by using a local variable here or just ignore ktlint if it's him that complains about.

**Review by 0xnm — APPROVED** [review]

**Review by xgouchet — APPROVED** [review]
